### ISSUE TITLE
 no-noninteractive-element-interactions rule is triggered with onSubmit event

### ISSUE THREAD

**Issue by fernandopasik** [evcohen/eslint-plugin-jsx-a11y#253]
With this example the rule triggers an error:

```
<form onSubmit={ this.handleSubmit }>
</form>
```

**Comment by axelboc** [comment]
I'm experiencing this issue with iframe and onLoad.

**Comment by axelboc** [comment]
The rule description suggests that the rule only applies to mouse and keyboard events, but this doesn't seem to be the case in practice:

> Non-interactive elements should not be assigned mouse or keyboard event listeners.

**Comment by ljharb** [comment]
onSubmit is both a mouse and keyboard event listener.

onLoad, I agree should be allowed on `img`, `iframe`, and `body` tags.

**Comment by axelboc** [comment]
Throwing my two cents about the form: if `onSubmit` event is considered a mouse and keyboard event listener, then shouldn't the `form` be considered as an interactive element?

**Comment by axelboc** [comment]
Throwing my two cents about the form: I don't see any accessibility bad practice in @fernandopasik's example. What about the `reset` event, or events that bubble up like `change` (I think)? Predicting whether a developer uses these events in a non-accessible manner seems way out of scope. I feel like the term "interactions" should only cover what browsers call "user gestures" (the gestures a browser requires to allow the use of some APIs, like the Full Screen API). If I'm not mistaken, these are simply mouse and keyboard events.

**Comment by fernandopasik** [comment]
I thought **onSubmit** and **onReset** are form events, not keyboard nor mouse events. @ljharb please could you share documentation regarding this?

**Comment by ljharb** [comment]
@fernandopasik you're right, they're form events, but form events result from user interaction (keyboard and/or mouse), so conceptually they apply here.

That said, I totally agree that `onSubmit` makes sense on `<form>` (and only on `<form>`), which is why I added the "bug" label immediately :-)

**Comment by fernandopasik** [comment]
@ljharb oh I'm sorry, I didn't notice the bug label! Thanks!

**Comment by jessebeach** [comment]
Agree. Good points. Will update the rule.

**Comment by jessebeach** [comment]
Fixed as of #281
